streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -440,7 +466,6 @@ public synchronized void start() throws IllegalStateException, StreamsException
 
         if (state == State.CREATED) {
             checkBrokerVersionCompatibility();
-            setState(State.RUNNING);
```

Comment:
Probably we should do a "tryTransitTo" function instead of a "check-and-set" pattern?